sqlglot/parser.py

```diff
@@ -1715,16 +1720,27 @@ def _parse_column_def(self, this):
             "unique": None,
             "parsed": True,
         }
+        constraint = None
 
         def parse_option(option, option_lambda):
+            nonlocal constraint
```

Comment:
why are you using a non local?

Reply:
ah because constraint is defined in 1723

Reply:
can you just use the options dictionary for this?

Reply:
no bc most entries in the dict can themselves be a named constraint